src/script/src/table.rs

```diff
@@ -48,6 +55,73 @@ pub struct ScriptsTable {
 }
 
 impl ScriptsTable {
+    /// this is used as a callback function when scripts table is created. `table` should be `scripts` table.
+    pub async fn recompile_register_udf(
+        table: TableRef,
+        query_engine: QueryEngineRef,
+    ) -> catalog::error::Result<()> {
+        let scan_stream = table
+            .scan(None, &[], None)
+            .await
+            .map_err(BoxedError::new)
+            .context(CompileScriptSnafu)?;
+        let ctx = SessionContext::new();
+        let rbs = scan_stream
+            .execute(0, ctx.task_ctx())
+            .map_err(BoxedError::new)
+            .context(CompileScriptSnafu)?;
+        let records = record_util::collect(rbs)
+            .await
+            .map_err(BoxedError::new)
+            .context(CompileScriptSnafu)?;
+        if records.is_empty() {
+            // scripts table is empty, no need to recompile
+            return Ok(());
+        }
+        assert_eq!(records.len(), 1);
```

Comment:
We can't ensure that there is only one batch. Let's iterate the batches and do the below stuff.